### DISCUSSION TITLE
Produce a DTO that is the difference between two objects.

### DISCUSSION THREAD

**Discussion by roxburghm** [riok/mapperly#1339] [Ideas]
Use mapperly to produce a DTO that has the properties set where they differ between two objects. 

So where you currently have

    Car DtoToCar(CarDto carDto);

And

    Void ApplyDto(CarDto carDto, Car targetCar);

I was thinking you could have

    CarDto GetDifferencesBetween(Car A, Car B);

This would be incredibly useful to generate "sparse" objects to patch databases or call APIs with minimum differences.

**Comment by latonz** [comment]
This doesn't seem to be a use-case for an object-to-object mapper and is therefore out of scope for Mapperly.
